dev/tools/gen_keycodes/README.md

```diff
@@ -0,0 +1,168 @@
+## Keycode Generator
+
+This directory contains a keycode generator that can generate Dart code for
+the `LogicalKeyboardKey` and `PhysicalKeyboardKey` classes. It draws information
+from both the Chromium and Android source bases, and incorporates the
+information it finds in those sources into a single key database in JSON form.
+
+It then generates `keyboard_key.dart` (containing the `LogicalKeyboardKey` and
+`PhysicalKeyboardKey` classes), and `keyboard_maps.dart`, containing
+platform-specific immutable maps for translating platform keycodes and
+information into the pre-defined key values in the `LogicalKeyboardKey` and
+`PhysicalKeyboardKey` classes. 
+
+The `data` subdirectory contains both some local data files, and the templates 
+used to generate the source files.
+
+ - `data/key_data.json`: contains the merged data from all the other sources.
+    This file will be regenerated if "--collect" is specified for the
+    gen_keycodes script.
+ - `data/key_name_to_android_name.json`: contains a mapping from Flutter key
+   names to Android keycode names (with the "KEY_" prefix stripped off).
+ - `data/keyboard_key.tmpl`: contains the template for the `keyboard_key.dart`
+   file. Markers that begin and end with "@@@" denote the locations where
+   generated data will be inserted.
+ - `data/keyboard_maps.tmpl`: contains the template for the `keyboard_maps.dart`
+   file. Markers that begin and end with "@@@" denote the locations where
+   generated data will be inserted.
+ - `data/printable.json`: contains a mapping between Flutter key name and its
+   printable character. This character is used as the key label.
+ 
+ ## Running the tool
+ 
+To run the `gen_keycodes` tool using the checked in `key_data.json` file, run
+it like so:
+
+```bash
+dart bin/gen_keycodes.dart
```

Comment:
`$flutter_root/bin/cache/dart-sdk/bin/dart` in the event that someone has a secondary dart installation on their path.

Reply:
Good point. Fixed.